airflow/models/taskinstance.py

```diff
@@ -3240,6 +3244,7 @@ def fetch_handle_failure_context(
             ti.state = State.UP_FOR_RETRY
             email_for_state = operator.attrgetter("email_on_retry")
             callbacks = task.on_retry_callback if task else None
+            ti.dag_run.deactivate_scheduling()
```

Comment:
Instead of "deactivating", can we set the next_schedulable to when the retry delay is over?

This does concern me a bit though. What if there are other tasks that aren't actually blocked by this task?

Reply:
Setting the date to the retry delay would be an issue since two tasks can go into retry at the same time with different retry delay. I feel the current method is better

Reply:
```
a (up for retry) -> b
c -> d
```

`a` being in up_for_retry shouldn't block the whole dagrun. It doesn't appear to (I have your branch running locally), but I wonder if that's a race condition...

Reply:
It shouldn't block it. `c` should be running; if `c` finishes, `d` should start and finish. The `b` should trigger once `a` retry_delay is reached